dandi/download.py

```diff
@@ -169,19 +169,13 @@ def download_generator(
                     dandiset["version"],
                     asset["uuid"],
                 )
-                if "sha256" not in asset:
-                    raise RuntimeError("sha256 hash not available for asset")
+                metadata = client.get_asset(*down_args)
+                for d in metadata.get("digest", []):
+                    if d["cryptoType"] == "dandi:SHA256":
+                        digests = {"sha256": d["value"]}
```

Comment:
sha256 is not guaranteed to be available, `dandi-etag` should always be there... we could test using both if present and even announce us whenever we run into a collision of having one matching and another not, but I would for now just switch to use `dandi-etag` consistently throughout and forget about sha256 as long as we are dealing with new API